ambari-server/src/main/java/org/apache/ambari/server/security/encryption/EncryptionUtils.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ambari.server.security.encryption;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.codec.binary.Base64;
+
+import com.google.inject.Singleton;
+
+@Singleton
+public class EncryptionUtils {
+
+  private static final String BASE_64_FIELD_DELIMITER = "::";
+  private static final String UTF_8_CHARSET = StandardCharsets.UTF_8.name();
+
+  private MasterKeyService environmentMasterKeyService;
+
+  /**
+   * Encrypts the given text using the master key found in the environment
+   * 
+   * @param toBeEncrypted
+   *          the text to be encrypted
+   * @return the String representation of the encrypted text
+   * @throws Exception
+   *           in case any error happened during the encryption process
+   */
+  public String encrypt(String toBeEncrypted) throws Exception {
+    return encrypt(toBeEncrypted, (String) null);
+  }
+
+  /**
+   * Encrypts the given text using the given master key
+   * 
+   * @param toBeEncrypted
+   *          the text to be encrypted
+   * @param masterKey
+   *          the mater key to be used for encryption
+   * @return the String representation of the encrypted text
+   * @throws Exception
+   *           in case any error happened during the encryption process
+   */
+  public String encrypt(String toBeEncrypted, String masterKey) throws Exception {
+    return encrypt(toBeEncrypted, getMasterKeyService(masterKey));
+  }
+
+  private MasterKeyService getMasterKeyService(String masterKey) {
+    if (masterKey == null) {
+      initEnvironmentMasterKeyService();
+      if (!environmentMasterKeyService.isMasterKeyInitialized()) {
+        throw new SecurityException("You are trying to use a persisted master key but its initialization has been failed!");
+      }
+      return environmentMasterKeyService;
+    }
+    return new MasterKeyServiceImpl(masterKey);
+  }
+
+  private void initEnvironmentMasterKeyService() {
+    if (environmentMasterKeyService == null) {
+      environmentMasterKeyService = new MasterKeyServiceImpl();
```

Comment:
should not we inject the instance that is used throughout application?

Reply:
It's not (yet) used by Guice.